In [1]:
import numpy as np
import pandas as pd

pd.set_option('display.max_columns', None) #show all columns

import warnings
warnings.simplefilter('ignore') #to avoid warnings

## Exploring the Data

In [2]:
sae=pd.read_csv('../data/sae_dataset.csv')
sae.head()

VehicleMarketCode  VehicleNationalCode VehicleSegmentationGlobal  \
0                ES                29263                 Small LCV   
1                ES                29263                 Small LCV   
2                ES                29263                 Small LCV   
3                ES                30020                         E   
4                ES                30020                         E   

  VehicleModelIntlName Bodytype_Group Fuel_Group Transmission_Group  \
0                Combo      Panel van     Diesel             Manual   
1                Combo      Panel van     Diesel             Manual   
2                Combo      Panel van     Diesel             Manual   
3                  607          Sedan     Diesel          Automated   
4                  607          Sedan     Diesel          Automated   

  Drivetype_Group  VehicleKW  VehicleDoors  VehicleWheelBase  VehicleSeats  \
0             2WD         73           4.0            2716.0             2   
1             2WD         73           4.0            2716.0             2   
2             2WD         73           4.0            2716.0             2   
3             2WD        150           4.0            2800.0             5   
4             2WD        150           4.0            2800.0             5   

   VehicleGrossNewPrice VehicleProductionStart VehicleProductionEnd  \
0               19321.0               7/1/2004            12/1/2011   
1               19321.0               7/1/2004            12/1/2011   
2               19321.0               7/1/2004            12/1/2011   
3               54250.0              12/1/2004             8/1/2012   
4               54250.0              12/1/2004             8/1/2012   

  VehicleMakeIntlName SAEReporteddate  StockDays PostCode registration_date  \
0     OPEL / VAUXHALL        2/3/2020         12    28983          7/1/2008   
1     OPEL / VAUXHALL        2/3/2020         86     8940         6/15/2010   
2     OPEL / VAUXHALL        2/3/2020        279     8840          7/1/2005   
3             PEUGEOT      10/12/2022         55    27002        10/15/2009   
4             PEUGEOT        2/3/2020         75    18015          7/1/2006   

   Mileage  Price  oricpric  PriceChangeCount  
0   238000   2990    2990.0               NaN  
1    90000   4499    4499.0               NaN  
2   210000   2490    2700.0               1.0  
3   259000   6500    6500.0               NaN  
4   155000   4900    7500.0               1.0

In [3]:
sae.shape

(192880, 24)

### 1. Descriptive stats

In order to get a basic understanding of the dataset, I explore the null values & data type in each column:

In [4]:
sae.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 192880 entries, 0 to 192879
Data columns (total 24 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   VehicleMarketCode          192880 non-null  object 
 1   VehicleNationalCode        192880 non-null  int64  
 2   VehicleSegmentationGlobal  192880 non-null  object 
 3   VehicleModelIntlName       192880 non-null  object 
 4   Bodytype_Group             192880 non-null  object 
 5   Fuel_Group                 192880 non-null  object 
 6   Transmission_Group         192880 non-null  object 
 7   Drivetype_Group            192880 non-null  object 
 8   VehicleKW                  192880 non-null  int64  
 9   VehicleDoors               192859 non-null  float64
 10  VehicleWheelBase           192834 non-null  float64
 11  VehicleSeats               192880 non-null  int64  
 12  VehicleGrossNewPrice       192880 non-null  float64
 13  VehicleProductionStart     19

In [5]:
null_cols=sae.isna().sum()
null_cols[null_cols>0]

VehicleDoors            21
VehicleWheelBase        46
oricpric               196
PriceChangeCount    119008
dtype: int64

In [6]:
null_cols=sae.isna().mean()*100
null_cols[null_cols>0]

VehicleDoors         0.010888
VehicleWheelBase     0.023849
oricpric             0.101618
PriceChangeCount    61.700539
dtype: float64

There are no relevant null values. Only the PriceChangeCount column has a high percentage 61.7% of null values. In normal circunstances I would delete the column but, in this case I will simply ignore it

##### Taking a look at the data type, it would be advisable to make some changes in order to perform possible calculations with them
* Natcode is the ID of AVG so, it should be a string instead of Integer
* Fields related to date are strings instead of datetime. I will set up as dates

In [7]:
sae['VehicleNationalCode']=sae['VehicleNationalCode'].astype('str')
sae['VehicleGrossNewPrice']=sae['VehicleGrossNewPrice'].astype('int')
sae['registration_date']=pd.to_datetime(sae['registration_date'], format='%m/%d/%Y')
sae['SAEReporteddate']=pd.to_datetime(sae['SAEReporteddate'], format='%m/%d/%Y')
sae['VehicleProductionStart']=pd.to_datetime(sae['VehicleProductionStart'], format='%m/%d/%Y')
sae['VehicleProductionEnd']=pd.to_datetime(sae['VehicleProductionEnd'], format='%m/%d/%Y', errors='coerce')

In [8]:
sae.dtypes

VehicleMarketCode                    object
VehicleNationalCode                  object
VehicleSegmentationGlobal            object
VehicleModelIntlName                 object
Bodytype_Group                       object
Fuel_Group                           object
Transmission_Group                   object
Drivetype_Group                      object
VehicleKW                             int64
VehicleDoors                        float64
VehicleWheelBase                    float64
VehicleSeats                          int64
VehicleGrossNewPrice                  int64
VehicleProductionStart       datetime64[ns]
VehicleProductionEnd         datetime64[ns]
VehicleMakeIntlName                  object
SAEReporteddate              datetime64[ns]
StockDays                             int64
PostCode                             object
registration_date            datetime64[ns]
Mileage                               int64
Price                                 int64
oricpric                        

### 2. Exploring the data in detail

#### 2.1 Categorical data 
I'm going to review the most relevant categorical columns from my point of view

In [9]:
sae.select_dtypes(exclude=['int64','float64'])

VehicleMarketCode VehicleNationalCode VehicleSegmentationGlobal  \
0                     ES               29263                 Small LCV   
1                     ES               29263                 Small LCV   
2                     ES               29263                 Small LCV   
3                     ES               30020                         E   
4                     ES               30020                         E   
...                  ...                 ...                       ...   
192875                PT              934825                         B   
192876                PT              934825                         B   
192877                PT              934825                         B   
192878                PT              934825                         B   
192879                PT              934964                         B   

       VehicleModelIntlName Bodytype_Group Fuel_Group Transmission_Group  \
0                     Combo      Panel van     Diesel             Manual   
1                     Combo      Panel van     Diesel             Manual   
2                     Combo      Panel van     Diesel             Manual   
3                       607          Sedan     Diesel          Automated   
4                       607          Sedan     Diesel          Automated   
...                     ...            ...        ...                ...   
192875                Micra      Hatchback     Petrol             Manual   
192876                Micra      Hatchback     Petrol             Manual   
192877                Micra      Hatchback     Petrol             Manual   
192878                Micra      Hatchback     Petrol             Manual   
192879                 Clio      Hatchback     Petrol             Manual   

       Drivetype_Group VehicleProductionStart VehicleProductionEnd  \
0                  2WD             2004-07-01           2011-12-01   
1                  2WD             2004-07-01           2011-12-01   
2                  2WD             2004-07-01           2011-12-01   
3                  2WD             2004-12-01           2012-08-01   
4                  2WD             2004-12-01           2012-08-01   
...                ...                    ...                  ...   
192875             2WD             2022-02-01                  NaT   
192876             2WD             2022-02-01                  NaT   
192877             2WD             2022-02-01                  NaT   
192878             2WD             2022-02-01                  NaT   
192879             2WD             2022-04-01                  NaT   

       VehicleMakeIntlName SAEReporteddate PostCode registration_date  
0          OPEL / VAUXHALL      2020-02-03    28983        2008-07-01  
1          OPEL / VAUXHALL      2020-02-03     8940        2010-06-15  
2          OPEL / VAUXHALL      2020-02-03     8840        2005-07-01  
3                  PEUGEOT      2022-10-12    27002        2009-10-15  
4                  PEUGEOT      2020-02-03    18015        2006-07-01  
...                    ...             ...      ...               ...  
192875              NISSAN      2022-10-12     1000        2022-05-15  
192876              NISSAN      2022-10-12     3515        2022-04-15  
192877              NISSAN      2022-10-12     8200        2022-03-15  
192878              NISSAN      2022-10-12     2830        2022-05-15  
192879             RENAULT      2022-10-12     2420        2022-05-15  

[192880 rows x 14 columns]

##### a few questions to learn more about the data
How many markets/records appear in the dataset?

In [10]:
sae.VehicleMarketCode.value_counts()

ES    176407
PT     16473
Name: VehicleMarketCode, dtype: int64

What are manufacturers that contain more records?

In [11]:
sae.VehicleMakeIntlName.value_counts()

RENAULT       23370
PEUGEOT       17070
SEAT          15328
BMW           14385
FORD (EUR)    14078
              ...  
PIAGGIO           2
MPM MOTORS        1
BORGWARD          1
MORGAN            1
SAAB              1
Name: VehicleMakeIntlName, Length: 61, dtype: int64

What is the most repeated Natcode?

In [12]:
sae.VehicleNationalCode.value_counts()

74488     852
83098     652
80819     569
68495     567
56630     541
         ... 
56877       1
56901       1
56903       1
83619       1
934964      1
Name: VehicleNationalCode, Length: 15867, dtype: int64

The most repeated vehicle is the next:

In [13]:
sae[sae.VehicleNationalCode=='74488'].head()

VehicleMarketCode VehicleNationalCode VehicleSegmentationGlobal  \
116852                ES               74488                     B SUV   
116853                ES               74488                     B SUV   
116854                ES               74488                     B SUV   
116855                ES               74488                     B SUV   
116856                ES               74488                     B SUV   

       VehicleModelIntlName Bodytype_Group Fuel_Group Transmission_Group  \
116852                 C-HR      Coupe SUV     Hybrid          Automated   
116853                 C-HR      Coupe SUV     Hybrid          Automated   
116854                 C-HR      Coupe SUV     Hybrid          Automated   
116855                 C-HR      Coupe SUV     Hybrid          Automated   
116856                 C-HR      Coupe SUV     Hybrid          Automated   

       Drivetype_Group  VehicleKW  VehicleDoors  VehicleWheelBase  \
116852             2WD         90           5.0            2640.0   
116853             2WD         90           5.0            2640.0   
116854             2WD         90           5.0            2640.0   
116855             2WD         90           5.0            2640.0   
116856             2WD         90           5.0            2640.0   

        VehicleSeats  VehicleGrossNewPrice VehicleProductionStart  \
116852             5                 28990             2016-12-01   
116853             5                 28990             2016-12-01   
116854             5                 28990             2016-12-01   
116855             5                 28990             2016-12-01   
116856             5                 28990             2016-12-01   

       VehicleProductionEnd VehicleMakeIntlName SAEReporteddate  StockDays  \
116852           2019-10-01              TOYOTA      2020-02-03         29   
116853           2019-10-01              TOYOTA      2020-02-03         17   
116854           2019-10-01              TOYOTA      2022-10-12          5   
116855           2019-10-01              TOYOTA      2022-10-12          2   
116856           2019-10-01              TOYOTA      2020-02-03         16   

       PostCode registration_date  Mileage  Price  oricpric  PriceChangeCount  
116852    29004        2018-07-01    38600  21990   21990.0               NaN  
116853    25194        2019-07-01     3720  22200   22200.0               NaN  
116854    46014        2019-09-15    47467  25750   25750.0               NaN  
116855    33600        2018-07-01    36000  25900   25900.0               NaN  
116856    46015        2019-07-01    11893  23900   23900.0               NaN

##### About date columns

As Carsten told me about Dataset, SAEReportedDate was the date the snippet was taken from. I'd like to check how many reports are on this dataset

In [14]:
sae.SAEReporteddate.value_counts()

2020-02-03    118597
2022-10-12     74283
Name: SAEReporteddate, dtype: int64

There are two reports in this dataset. Perhaps it is interesting to take it into account separately, since there have been external circumstances that have been able to alter the values (e.g. the global pandemic)

In [15]:
sae2020=sae[sae.SAEReporteddate=='2020-02-03']
sae2022=sae[sae.SAEReporteddate=='2022-10-12']

It may be interesting to create some additional columns in the dataset:
* New column with the **"age of the vehicle"** according to its **registration_date** until the respective **SAEReportedDate**
* New column with the **"loss of value"** of the vehicle according to **VehicleGrossNewPrice** and his current **Price**

In [16]:
sae['AgeOfVeh']=(sae['SAEReporteddate'] - sae['registration_date'])/np.timedelta64(1,'D')
sae['LossVal']=(sae['VehicleGrossNewPrice']-sae['Price'])

Based on that, also it is interesting to know the next variables:
* The relation between **"loss of value"** according to the **Mileage** in order to know the devaluation per distance
* The relation between **"loss of value"** according to the **AgeOfVeh** in order to know the devaluation per time
* The relation between **"Mileage"** according to the **AgeOfVeh** in order to know the distance per time

In [17]:
sae['DevPerKm']=(sae['LossVal']/sae['Mileage'])
sae['DevPerAge']=(sae['LossVal']/sae['AgeOfVeh'])
sae['KmPerAge']=(sae['Mileage']/sae['AgeOfVeh'])

In [18]:
sae.pivot_table(index=['VehicleMarketCode'], columns=['SAEReporteddate'], aggfunc='mean')

AgeOfVeh               DevPerAge              DevPerKm  \
SAEReporteddate    2020-02-03   2022-10-12 2020-02-03 2022-10-12 2020-02-03   
VehicleMarketCode                                                             
ES                 1412.12425  1897.737191  13.419215   6.645931   1.373761   
PT                 1431.61314  1946.491854   8.795999   4.610226   0.314525   

                               KmPerAge                  LossVal  \
SAEReporteddate   2022-10-12 2020-02-03 2022-10-12    2020-02-03   
VehicleMarketCode                                                  
ES                  0.408987  49.388000  47.210113  12938.089638   
PT                  0.141568  59.018516  53.215307  10093.014350   

                                      Mileage                        Price  \
SAEReporteddate      2022-10-12    2020-02-03     2022-10-12    2020-02-03   
VehicleMarketCode                                                            
ES                 10699.460478  68701.517088   86884.679629  16832.019462   
PT                  8610.318741  84081.331357  101563.574058  15039.743952   

                                PriceChangeCount             StockDays  \
SAEReporteddate      2022-10-12       2020-02-03 2022-10-12 2020-02-03   
VehicleMarketCode                                                        
ES                 21377.179025         1.938592   1.850572  78.239822   
PT                 18750.737097         1.812662   1.616865  86.357922   

                             VehicleDoors            VehicleGrossNewPrice  \
SAEReporteddate   2022-10-12   2020-02-03 2022-10-12           2020-02-03   
VehicleMarketCode                                                           
ES                 72.838018     4.685257   4.717294         29770.109100   
PT                 91.244373     4.811670   4.821415         25132.758302   

                                 VehicleKW             VehicleSeats  \
SAEReporteddate      2022-10-12 2020-02-03  2022-10-12   2020-02-03   
VehicleMarketCode                                                     
ES                 32076.639504  98.361067  103.565774     4.911097   
PT                 27361.055839  77.319616   85.251834     4.918287   

                             VehicleWheelBase                   oricpric  \
SAEReporteddate   2022-10-12       2020-02-03   2022-10-12    2020-02-03   
VehicleMarketCode                                                          
ES                   4.88933      2673.098723  2675.497191  17186.069278   
PT                   4.93471      2624.070090  2637.866683  15339.980313   

                                 
SAEReporteddate      2022-10-12  
VehicleMarketCode                
ES                 21664.264558  
PT                 18777.759483

#### 2.2 Numerical data
In order to look at multiple descriptive statistics at once, I use the describe function. This will show the count, mean, standard deviation, minimum, 25th percentile, median (50th percentile), 75th percentile, and the maximum.

In [19]:
sae.describe().T

count          mean           std           min  \
VehicleKW             192880.0     98.682186     47.152377      4.000000   
VehicleDoors          192859.0      4.707465      0.696083      2.000000   
VehicleWheelBase      192834.0   2670.309411    180.183084   1686.000000   
VehicleSeats          192880.0      4.904920      0.725362      1.000000   
VehicleGrossNewPrice  192880.0  30259.096283  18962.733322   7140.000000   
StockDays             192880.0     77.281688     75.502168      0.000000   
Mileage               192880.0  76988.615009  57069.621542      1.000000   
Price                 192880.0  18394.630418  13789.559508   1390.000000   
oricpric              192684.0  18706.690836  16686.700415     -1.000000   
PriceChangeCount       73872.0      1.889173      1.773183      1.000000   
AgeOfVeh              192880.0   1602.030682   1030.650527     19.000000   
LossVal               192880.0  11864.465865   9829.591844 -24251.000000   
DevPerKm              192880.0      0.944747     70.636184  -4232.000000   
DevPerAge             192880.0     10.523674     14.271578   -424.242424   
KmPerAge              192880.0     49.220602     26.860720      0.017241   

                               25%           50%            75%           max  
VehicleKW                73.000000     88.000000     110.000000  5.860000e+02  
VehicleDoors              5.000000      5.000000       5.000000  6.000000e+00  
VehicleWheelBase       2580.000000   2648.000000    2728.000000  4.522000e+03  
VehicleSeats              5.000000      5.000000       5.000000  9.000000e+00  
VehicleGrossNewPrice  20450.000000  25850.000000   33390.000000  4.072500e+05  
StockDays                19.000000     53.000000     110.000000  4.010000e+02  
Mileage               27780.750000  69000.000000  113727.750000  3.500000e+05  
Price                 10990.000000  15400.000000   21028.000000  4.159000e+05  
oricpric              11100.000000  15500.000000   21500.000000  2.390000e+06  
PriceChangeCount          1.000000      1.000000       2.000000  9.500000e+01  
AgeOfVeh                833.000000   1550.000000    2120.000000  5.841000e+03  
LossVal                6160.000000   9605.000000   14685.250000  2.196100e+05  
DevPerKm                  0.095000      0.152639       0.290000  2.822000e+04  
DevPerAge                 4.461573      7.308266      12.268041  1.628947e+03  
KmPerAge                 30.786662     44.856766      62.829989  4.578947e+02

The two main measures of central tendency are the mean and the median (50th percentile). Those variables that are close in mean and median value, means the data is symmetrically distributed around the mean.

On most important fields considered by me for the dataset, I'm seeing the mean is greater than the median, so data is right skewed. This means there are a few outliers that influence the mean.

#### Cleaning some typing errors

Quick look to oricpric column where the max value is highest than VehicleGrossNewPrice

In [20]:
sae[(sae.oricpric==2390000)]

VehicleMarketCode VehicleNationalCode VehicleSegmentationGlobal  \
165374                ES               85177                     C SUV   

       VehicleModelIntlName Bodytype_Group Fuel_Group Transmission_Group  \
165374             Sportage     Closed SUV     Diesel             Manual   

       Drivetype_Group  VehicleKW  VehicleDoors  VehicleWheelBase  \
165374             2WD         85           5.0            2670.0   

        VehicleSeats  VehicleGrossNewPrice VehicleProductionStart  \
165374             5                 32200             2019-05-01   

       VehicleProductionEnd VehicleMakeIntlName SAEReporteddate  StockDays  \
165374           2020-10-01                 KIA      2022-10-12        210   

       PostCode registration_date  Mileage  Price   oricpric  \
165374     8201        2020-01-15    22500  26900  2390000.0   

        PriceChangeCount  AgeOfVeh  LossVal  DevPerKm  DevPerAge   KmPerAge  
165374               3.0    1001.0     5300  0.235556   5.294705  22.477522

##### Could be some typing errors in oricpric column that could be fixed before analysis. Lets take a look from highest to lowest sort values

In [21]:
sae.sort_values(by=['oricpric'], ascending=False)

VehicleMarketCode VehicleNationalCode VehicleSegmentationGlobal  \
165374                ES               85177                     C SUV   
104344                ES               72203                 Large LCV   
170239                ES               89414                         B   
89277                 ES               69004                     Sport   
89274                 ES               69004                     Sport   
...                  ...                 ...                       ...   
174971                ES               93204                     C SUV   
175190                ES               93519                         C   
175252                ES               93728                         B   
175747                ES               93917                         C   
175924                ES               94315                         C   

       VehicleModelIntlName Bodytype_Group      Fuel_Group Transmission_Group  \
165374             Sportage     Closed SUV          Diesel             Manual   
104344             Boxer 35      Panel van          Diesel             Manual   
170239                 Clio      Hatchback          Hybrid          Automated   
89277             Aventador         Cabrio          Petrol          Automated   
89274             Aventador         Cabrio          Petrol          Automated   
...                     ...            ...             ...                ...   
174971            Formentor     Closed SUV  Plug-in Hybrid          Automated   
175190            Megane IV      Hatchback  Plug-in Hybrid          Automated   
175252                Ibiza      Hatchback             CNG             Manual   
175747                  308      Hatchback          Diesel          Automated   
175924                 Leon      Hatchback             CNG             Manual   

       Drivetype_Group  VehicleKW  VehicleDoors  VehicleWheelBase  \
165374             2WD         85           5.0            2670.0   
104344             2WD         96           4.0            4035.0   
170239             2WD        103           5.0            2583.0   
89277              4WD        515           2.0            2700.0   
89274              4WD        515           2.0            2700.0   
...                ...        ...           ...               ...   
174971             2WD        180           5.0            2680.0   
175190             2WD        117           5.0            2670.0   
175252             2WD         66           5.0            2564.0   
175747             2WD         96           5.0            2675.0   
175924             2WD         96           5.0            2686.0   

        VehicleSeats  VehicleGrossNewPrice VehicleProductionStart  \
165374             5                 32200             2019-05-01   
104344             3                 35950             2016-05-01   
170239             5                 23460             2020-06-01   
89277              2                407250             2015-01-01   
89274              2                407250             2015-01-01   
...              ...                   ...                    ...   
174971             5                 48840             2021-03-01   
175190             5                 41425             2021-05-01   
175252             5                 21960             2021-06-01   
175747             5                 33680             2021-06-01   
175924             5                 27010             2021-08-01   

       VehicleProductionEnd VehicleMakeIntlName SAEReporteddate  StockDays  \
165374           2020-10-01                 KIA      2022-10-12        210   
104344           2019-05-01             PEUGEOT      2022-10-12          5   
170239           2022-01-01             RENAULT      2022-10-12        227   
89277            2019-02-01         LAMBORGHINI      2020-02-03         56   
89274            2019-02-01         LAMBORGHINI      2020-02-03        261   
...                

In [22]:
sae['oricpric']=sae['oricpric'].replace(to_replace=2390000, value=23900)
sae['oricpric']=sae['oricpric'].replace(to_replace=2350000, value=23500)
sae['oricpric']=sae['oricpric'].replace(to_replace=2070000, value=20700)

Checking the "oricpric" column in ascending order, it's not possible to fix more data in this approximation.

In [23]:
sae.sort_values(by=['oricpric'], ascending=True)

VehicleMarketCode VehicleNationalCode VehicleSegmentationGlobal  \
177519                PT               56530                         A   
181620                PT               72456                     B SUV   
176873                PT               53118                         C   
179093                PT               64600                         C   
180310                PT               68121                         C   
...                  ...                 ...                       ...   
174971                ES               93204                     C SUV   
175190                ES               93519                         C   
175252                ES               93728                         B   
175747                ES               93917                         C   
175924                ES               94315                         C   

        VehicleModelIntlName Bodytype_Group      Fuel_Group  \
177519                   i10      Hatchback          Petrol   
181620             C4 Cactus     Closed SUV          Diesel   
176873             Giulietta      Hatchback          Diesel   
179093                  Leon      Hatchback          Diesel   
180310  Megane III Grandtour   Stationwagon          Diesel   
...                      ...            ...             ...   
174971             Formentor     Closed SUV  Plug-in Hybrid   
175190             Megane IV      Hatchback  Plug-in Hybrid   
175252                 Ibiza      Hatchback             CNG   
175747                   308      Hatchback          Diesel   
175924                  Leon      Hatchback             CNG   

       Transmission_Group Drivetype_Group  VehicleKW  VehicleDoors  \
177519             Manual             2WD         62           5.0   
181620             Manual             2WD         73           5.0   
176873             Manual             2WD         77           5.0   
179093             Manual             2WD        110           5.0   
180310             Manual             2WD         81           5.0   
...                   ...             ...        ...           ...   
174971          Automated             2WD        180           5.0   
175190          Automated             2WD        117           5.0   
175252             Manual             2WD         66           5.0   
175747          Automated             2WD         96           5.0   
175924             Manual             2WD         96           5.0   

        VehicleWheelBase  VehicleSeats  VehicleGrossNewPrice  \
177519            2380.0             5                 11200   
181620            2595.0             5                 24360   
176873            2634.0             5                 25600   
179093            2634.0             5                 32666   
180310            2703.0             5                 28180   
...                  ...           ...                   ...   
174971            2680.0             5                 48840   
175190            2670.0             5                 41425   
175252            2564.0             5                 21960   
175747            2675.0             5                 33680   
175924            2686.0             5                 27010   

       VehicleProductionStart VehicleProductionEnd VehicleMakeIntlName  \
177519             2011-03-01           2012-11-01             HYUNDAI   
181620             2014-11-01           2018-05-01             CITROEN   
176873             2010-04-01           2013-03-01          ALFA ROMEO   
179093             2013-01-01           2015-06-01                SEAT   
180310             2013-10-01           2015-05-01             RENAULT   
...                       ...                  ...                 ...   
174971             2021-03-01                  NaT               CUPRA   
175190             2021-05-01           2022-04-01             RENAULT   
175252             2021-06-01           2022-06-01                SEAT   
175747     

#### Save the dataset updated

In [24]:
sae.to_csv('../data/sae_dataset_updated.csv', index=False)